Packs/DeCYFIR/Integrations/decyfiriocs/decyfiriocs.py

```diff
@@ -0,0 +1,499 @@
+                                    data1["labels"] = raw_ta_obj.get("labels", [])
+                            raw_ta_data[data1.get(LABEL_ID)] = data1
+
+                    # Mapping the relations with source and target objects
+                    if raw_ta_rels is not None and raw_ta_rels:
+                        for raw_ta_rel_ in raw_ta_rels:
+                            # Source ref obj from relationship obj
+                            if raw_ta_rel_.get(LABEL_SOURCE_REF) in raw_ta_data:
+                                source_ref_obj: Dict = raw_ta_data.get(raw_ta_rel_.get(LABEL_SOURCE_REF, ''), {})
+                            else:
+                                source_ref_obj = raw_ta_data.get(raw_ta_rel_.get('sourceRef', ''), {})
+                            # Target ref obj from relationship obj
+                            if raw_ta_rel_.get(LABEL_TARGET_REF) in raw_ta_data:
+                                target_ref_obj: Dict = raw_ta_data.get(raw_ta_rel_.get(LABEL_TARGET_REF, ''), {})
+                            else:
+                                target_ref_obj = raw_ta_data.get(raw_ta_rel_.get('targetRef', ''), {})
+
+                            if source_ref_obj is not None and target_ref_obj is not None:
+                                if raw_ta_obj.get(LABEL_ID) != source_ref_obj.get(LABEL_ID):
+                                    source_ti_data_obj = self.build_threat_intel_indicator_obj(source_ref_obj, tlp_color, feed_tags)
+                                    if source_ti_data_obj is not None and source_ti_data_obj:
+                                        return_data.append(source_ti_data_obj)
+                                else:
+                                    source_ti_data_obj = ta_source_obj
+
+                                if raw_ta_obj.get(LABEL_ID) != target_ref_obj.get(
+                                    LABEL_ID) and source_ti_data_obj is not None and source_ti_data_obj.get(LABEL_ID) != target_ref_obj.get(LABEL_ID):
+
+                                    target_ti_data_obj = self.build_threat_intel_indicator_obj(target_ref_obj, tlp_color, feed_tags)
+                                    if target_ti_data_obj is not None and target_ti_data_obj:
+                                        return_data.append(target_ti_data_obj)
+                                else:
+                                    target_ti_data_obj = ta_source_obj
+
+                                if source_ti_data_obj and target_ti_data_obj:
+                                    if raw_ta_obj.get(LABEL_ID) != source_ref_obj.get(LABEL_ID):
+                                        ti_relationships: dict = self.build_threat_actor_relationship_obj(source_ti_data_obj, target_ti_data_obj)
+                                        source_ti_data_obj[LABEL_RELATIONSHIPS] = []
+                                        if ti_relationships:
+                                            if source_ti_data_obj[LABEL_RELATIONSHIPS]:
```

Comment:
in line 300: `source_ti_data_obj[LABEL_RELATIONSHIPS] = []`
so when it will reach the if body?